src/menu_events.cpp

```diff
@@ -454,6 +454,39 @@ void menu_handler::show_enemy_moves(bool ignore_units, int side_num)
 			gui_->highlight_another_reach(path);
 		}
 	}
+	
+	//   Fetch cursor position, in order to later find a possible
+	// unit under it.
+	int x = -1;
+	int y = -1;
+	// auto bitmask =
+			SDL_GetMouseState(&x, &y);
+
+	//   Find possible unit (no matter whether friend or foe) under
+	// the cursor.
+	const mouse_handler& _mouse_handler =
+			const_cast<mouse_handler&>(
+					pc_.get_mouse_handler_base());
+	const game_display& _gd =
+			const_cast<game_display&>(
+					const_cast<mouse_handler&>(
+							_mouse_handler
+					).gui____public_for_demo____REMOVEME());
+	const map_location _hex_under_mouse = _gd.hex_clicked_on(x, y);
+	const auto _selected_hex_has_unit = _mouse_handler
+			.find_unit____public_for_demo____REMOVEME(
+					_hex_under_mouse
+			).valid();
```

Comment:
Similarly, it seems to me that this visibility problem could be solved by adding a function to `mouse_handler` returning bool.

Reply:
From my beginner perspective I find it would be acceptable to change the visibility of `unit_map::const_iterator mouse_handler::find_unit(const map_location&) const` from `protected` to `public`.

Will add instead a `public` function `bool mouse_handler::hex_hosts_unit(const map_location&) const` wrapping a call to `iterator_base<unit_map::standard_iter_types>.valid()` on the result of `unit_map::iterator mouse_handler::find_unit(const map_location&)` given my understanding of that you are more appealed by this.unit(const map_location&)` given my understanding of that you are more appealed by this.

Reply:
Well, I'm not opposed to making `find_unit` public; I was just considering that what you need here is merely whether a unit is there, not a reference to the unit if present.